src/main/java/com/team766/controllers/CanSparkMaxSmartMotionPIDController.java

```diff
@@ -24,61 +24,61 @@
 import com.team766.library.RateLimiter;
 
 
-public class CSMSMPID{
+public class CanSparkMaxSmartMotionPIDController{
 	// The attributes of the class include references to the motor controller, SparkMax controller, PID controller, and absolute encoder
-	private MotorController mc1;
-	private CANSparkMax csm1;
-	private SparkMaxPIDController pid1;
-	private SparkMaxAbsoluteEncoder abs1;
+	private MotorController mc;
+	private CANSparkMax csm;
+	private SparkMaxPIDController pid;
+	private SparkMaxAbsoluteEncoder abs;
 	//PID Related Variables
 	private static double dz1 = 0; 
-	private static double maxpos1 = 0;
-	private static double minpos1 = 0;
-	private static double maxvel1 = 0;
-	private static double maxaccel1 = 0;
-	private static double maxspeed1 = 0;
-	private static double minspeed1 = 0;
-	private static double currentPos = 0;
-	private static double combo;
-
-	//antigrav variable
+	private static double setPointPosition = 0;
+	private static double comboOfTimesInsideDeadzone = 0;
+	private static double minPos = 0;
+	private static double maxPos = 0;
+
+	private bool isAbsoluteEncoderEnabled;
+
+	//antigrav coefficient
 	private static double antiGravK;
 	//enum for which state the PID is in
 	public enum PIDSTATE{
 		PID,
 		OFF,
 		ANTIGRAV
 	}
-	//the state of the PID2
+	//the state of the PID
 	private PIDSTATE theState = PIDSTATE.OFF;
 	
 	//constructor for the class with no absolute encoder
-	public CSMSMPID(String configName) throws Exception{
-			//loggerCategory = Category.MECHANISMS;
+	public CanSparkMaxSmartMotionPIDController(String configName) throws RuntimeException{
```

Comment:
remove the `throws` declaration - the point of RuntimeException is that you don't need to declare it
(here and in the constructor below)